tests/test_caching.py

```diff
@@ -0,0 +1,52 @@
+#    Copyright 2015-2017 ARM Limited
```

Comment:
Hehe will fix.. infact is it Ok to make this "ARM Limited, Google and Contributors" even?